### ISSUE TITLE
A11y: Time range form - Implicit Submission

### ISSUE THREAD

**Issue by matejkubinec** [grafana/grafana#52577]
**Steps to reproduce**:

1. Open a time range form (Dashboards - from/to range or Silence interval in Alerts)
2. Change a value in the from/to input field
3. With the input field still focused hit enter

**Actual Result**:

No change happens.

**Expected Result**

The form should be submitted and the time range changed.

**Relevant WCAG Criteria:** [2.1 Keyboard](https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0#qr-keyboard-operation-keyboard-operable)

**Environment**:
- Grafana version: 9.1.0-pre
- Data source type & version: -
- User OS & Browser: MacOS Big Sur, Chrome 103
- Others:
